pkg/cloud/services/iam/podidentitywebhook.go

```diff
@@ -0,0 +1,340 @@
+		RoleRef: v12.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     podIdentityWebhookName,
+		},
+		Subjects: []v12.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      podIdentityWebhookName,
+				Namespace: ns,
+			},
+		},
+	}
+
+	return remoteClient.Create(ctx, crb)
+}
+
+func reconcileService(ctx context.Context, ns string, remoteClient client.Client) error {
+	check := &corev1.Service{}
+	if err := remoteClient.Get(ctx, types.NamespacedName{
+		Name:      podIdentityWebhookName,
+		Namespace: ns,
+	}, check); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if check.UID != "" {
+		return nil
+	}
+	service := &corev1.Service{
+		ObjectMeta: objectMeta(podIdentityWebhookName, ns),
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port:       443,
+					TargetPort: intstr.FromInt(443),
+				},
+			},
+			Selector: map[string]string{
+				"app": podIdentityWebhookName,
+			},
+		},
+	}
+
+	return remoteClient.Create(ctx, service)
+}
+
+func reconcileDeployment(ctx context.Context, ns string, secret *corev1.Secret, remoteClient client.Client) error {
+	check := &v13.Deployment{}
+	if err := remoteClient.Get(ctx, types.NamespacedName{
+		Name:      podIdentityWebhookName,
+		Namespace: ns,
+	}, check); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if check.UID != "" {
+		return nil
+	}
+
+	replicas := int32(1)
+	deployment := &v13.Deployment{
+		ObjectMeta: objectMeta(podIdentityWebhookName, ns),
+		Spec: v13.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": podIdentityWebhookName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": podIdentityWebhookName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: podIdentityWebhookName,
+					Containers: []corev1.Container{
+						{
+							Name:            podIdentityWebhookName,
+							Image:           podIdentityWebhookImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "webhook-certs",
+									MountPath: "/etc/webhook/certs/",
+									ReadOnly:  false,
+								},
+							},
+							Command: []string{
+								"/webhook",
+								"--in-cluster=false",
+								"--namespace=" + ns,
+								"--service-name=" + podIdentityWebhookName,
+								"--annotation-prefix=eks.amazonaws.com",
+								"--token-audience=sts.amazonaws.com",
+								"--logtostderr",
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "webhook-certs",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: secret.Name,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return remoteClient.Create(ctx, deployment)
```

Comment:
I think this should be a create or update or something like that if this is really a reconciliation process. Right?

Reply:
most of these reconcilers basically make sure the thing exists or creates it... there is no check if anything changed and to fix it via an update back to what we need... that makes it a bit more complex but if we want that on all of these i can try one and see if we like the direction and then implement the rest

Reply:
Sounds good!